lib/widgets/favorite_filter.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:flutter/material.dart';
+import 'package:hive/hive.dart';
+
+import '../constants.dart';
+import '../domain/favorite_model.dart';
+import '../domain/hive_model.dart';
+
+class FavoriteFilterWidget extends StatefulWidget {
+  const FavoriteFilterWidget({super.key});
+
+  @override
+  State<FavoriteFilterWidget> createState() => _FavoriteFilterWidgetState();
+}
+
+class _FavoriteFilterWidgetState extends State<FavoriteFilterWidget> {
+  final favoriteResultBox = Hive.box<ShowHive>('favoriteScreenBox');
+  final favoriteBox = Hive.box<ShowHive>('favoriteLocalBox');
+  final filterBox = Hive.box<ShowHive>('filterBox');
+  final favoriteFilter = FirecloudeEssense();
```

Comment:
Consider either making this entity a **singleton** or injecting it in the init function when the app starts.

Reply:
Done, im used get_it package.